### PR TITLE
Add Linux to platform channel example

### PR DESCRIPTION
**Pull request by stuartmorgan**
This adds a Linux section to the example implementation instructions, following the same structure as the existing examples, and using the code from https://github.com/flutter/flutter/pull/108068

This is the Linux equivalent of https://github.com/flutter/website/pull/7328

Part of https://github.com/flutter/flutter/issues/79204

### CODE REVIEW

**Comment by stuartmorgan** [comment]
I'm just submitting this now this since it's ready and I don't want to lose the edits; this can wait for after the freeze.

File: src/development/platform-integration/platform-channels.md [hunk 1]
```diff
@@ -205,14 +205,15 @@ platform side and vice versa:
 | Map                        | FlValue(FlValue, FlValue) |
 {% endsamplecode %}
 
-## Example: Calling platform-specific Android, iOS, and Windows code using platform channels {#example}
+## Example: Calling platform-specific Android, iOS, Windows, and Linux code using platform channels {#example}
```
Comment (loic-sharma) [inline src/development/platform-integration/platform-channels.md:208]: This heading is getting really long. Should we simplify it?

```suggestion
## Example: Calling platform-specific code using platform channels {#example}
```

File: src/development/platform-integration/platform-channels.md [hunk 2]
```diff
@@ -872,6 +873,142 @@ And replace with the following:
...
+```c
+static void my_application_dispose(GObject* object) {
+  MyApplication* self = MY_APPLICATION(object);
+  g_clear_pointer(&self->dart_entrypoint_arguments, g_strfreev);
+  g_clear_object(&self->battery_channel);
+  G_OBJECT_CLASS(my_application_parent_class)->dispose(object);
+}
+```
+
+Edit the `my_application_activate` method and initilize
```
Comment (loic-sharma) [inline src/development/platform-integration/platform-channels.md:930]:
```suggestion
Edit the `my_application_activate` method and initialize
```

**Review by loic-sharma — APPROVED** [review]

File: src/development/platform-integration/platform-channels.md (hunk 1 above)
Comment (stuartmorgan) [reply src/development/platform-integration/platform-channels.md:208]: SGTM

File: src/development/platform-integration/platform-channels.md [hunk 3]
```diff
@@ -872,6 +873,142 @@ And replace with the following:
 You should now be able to run the application on Windows.
 If your device doesn't have a battery,
 it displays 'Battery level not available'.
+  
+### Step 6: Add a Linux platform-specific implementation
+  
+For this example you will need the `upower` developer headers installed.
```
Comment (sfshaza2) [inline src/development/platform-integration/platform-channels.md:879]:
```suggestion
For this example you need to install the `upower` developer headers.
```

File: src/development/platform-integration/platform-channels.md [hunk 4]
```diff
@@ -872,6 +873,142 @@ And replace with the following:
...
+Start by opening the Linux host portion of your Flutter app in the editor
+of your choice. The instructions below are for Visual Studio Code with the
+“C/C++” and “CMake” extensions installed, but can be adjusted for other IDEs.
+
+1. Launch Visual Studio Code.
+
+1. Open the **linux** directory inside your project.
+
+1. Choose **Yes** in the prompt asking: `Would you like to configure project "linux"?`.
+   This will allow C++ autocomplete to work.
```
Comment (sfshaza2) [inline src/development/platform-integration/platform-channels.md:894]:
```suggestion
   This enables C++ autocomplete.
```

File: src/development/platform-integration/platform-channels.md [hunk 5]
```diff
@@ -872,6 +873,142 @@ And replace with the following:
...
+<!--code-excerpt "my_application.cc" title-->
+```c
+struct _MyApplication {
+  GtkApplication parent_instance;
+  char** dart_entrypoint_arguments;
+  FlMethodChannel* battery_channel;
+};
+```
+
+Make sure it is cleaned up in `my_application_dispose`
```
Comment (sfshaza2) [inline src/development/platform-integration/platform-channels.md:918]:
```suggestion
Make sure to clean it up in `my_application_dispose`:
```

**Review by sfshaza2 — CHANGES_REQUESTED** [review]
A few tweaks, but otherwise lgtm

**Review by sfshaza2 — APPROVED** [review]
lgtm

**Pull request merged**